csrc/quantization/gptq_marlin/marlin_dtypes.cuh

```diff
@@ -54,7 +58,7 @@ class ScalarType<nv_bfloat16> {
   using FragS = Vec<nv_bfloat162, 1>;
   using FragZP = Vec<nv_bfloat162, 4>;
 
-#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 800
+#if !defined(__CUDA_ARCH__) || __CUDA_ARCH__ >= 800
```

Comment:
Why if not defined cuda arch? Is this needed for AMD?

Reply:
In host part (cpu part) compilation, `__CUDA_ARCH__ ` is not defined.

My code requires the bf16 num conversion functions are available for cpu code part.